llvm/test/CodeGen/AMDGPU/GlobalISel/legalize-ashr.mir

```diff
@@ -969,6 +969,176 @@ body: |
 
 ...
 
+---
```

Comment:
I didn't mean switch this to a MIR test. This should still be and end to end IR test

Reply:
It is testing the legalizer so it doesn't have to?

Reply:
At this point we are over-using MIR tests which are bigger, less maintainable, and test less. They are useful for testing edge cases that are difficult to reproduce, but simple stuff like this are better off just using IR. MIR legalizer tests were more useful at the start when we couldn't run as much code through end to end

Reply:
The benefit of using MIR here is to be able to get auto generate check lines. Maybe we can teach the script to be able to generate check lines for llc test with IR input and MIR output.

Reply:
You can generate ISA checks just fine, and update_llc_test_checks works much better than update_mir_test_checks